File: Backtracking/python/15657.py
n,m = map(int,input().split())
arr = sorted(map(int,input().split()))

answer = []
def combinations (idx,result,length):
  if length == m:
    answer.append(" ".join(map(str,result)))
    return
  for i in range(idx,n):
    combinations(i,result+[arr[i]],length+1)

for i in range(n):
  combinations(i,[arr[i]],1)
print("\n".join(answer))

File: Backtracking/javascript/15657.js
const input = require("fs").readFileSync("/dev/stdin").toString().split("\n");
const [n, m] = input[0].split(" ").map(Number);
const arr = input[1]
  .split(" ")
  .map(Number)
  .sort((a, b) => a - b);

const generate = () => {
  const answer = [];
  const combinations = (result, idx, len) => {
    if (len === m) {
      answer.push(result);
      return;
    }
    for (let i = idx; i < n; i++) {
      combinations(result + arr[i] + " ", i, len + 1);
    }
  };
  combinations("", 0, 0);
  return answer.join("\n");
};

const result = generate();
console.log(result);
